Libraries/Opc.Ua.Client/Session/Session.cs

```diff
@@ -2892,6 +2911,76 @@ public void ReadValues(
             }
         }
 
+        /// <inheritdoc/>
+        public byte[] ReadByteStringInChunks(NodeId nodeId)
+        {
+            if(ServerMaxByteStringLength <= 1)
+            {
+                throw new ServiceResultException(StatusCodes.BadIndexRangeNoData, "Server side MaxByteStringLength is not known or too small for reading data in chunks.");
+            }
+            int offset = 0;
+            List<byte[]> bytes = new List<byte[]>();
+
+            while(true)
+            {
+                ReadValueId valueToRead = new ReadValueId {
+                    NodeId = nodeId,
+                    AttributeId = Attributes.Value,
+                    IndexRange = new NumericRange(offset, offset + (int)ServerMaxByteStringLength - 1).ToString(),
```

Comment:
Doesn't this max value also depend on the message size configured by the client?